src/ui/pages/Notifications/NotificationItem.tsx

```diff
@@ -18,66 +20,87 @@ const NotificationItem = ({
 }: {
   item: KeriaNotification;
   index: number;
-  handleNotificationClick: (value: string) => void;
+  handleNotificationClick: (item: KeriaNotification) => void;
 }) => {
   const connectionsCache = useAppSelector(getConnectionsCache);
+  const [notificationLabelText, setNotificationLabelText] =
+    useState<string>("");
+  const [loading, setLoading] = useState(true);
+  const connection = connectionsCache.filter(
+    (connection) => connection.id === item.connectionId
+  )[0]?.label;
+  const [multisigIcpDetails, setMultisigIcpDetails] =
+    useState<MultiSigIcpRequestDetails | null>(null);
 
-  const notificationLabel = (item: KeriaNotification) => {
-    const connection = connectionsCache.filter(
-      (connection) => connection.id === item.connectionId
-    )[0]?.label;
-    // TODO: Implement different credential types
-    const credential = i18n.t(
-      "notifications.tab.credentialtypes.driverslicence"
+  const getMultisigIcpDetails = async () => {
+    const details = await Agent.agent.multiSigs.getMultisigIcpDetails(
+      item.a.d as string
     );
+    setMultisigIcpDetails(details);
+  };
+
+  const fetchNotificationLabel = async () => {
+    const label = await notificationLabel(item);
+    setNotificationLabelText(label);
+    setLoading(false);
```

Comment:
Hmm, will this run when the component is loaded, and then again when `multisigIcpDetails` is properly loaded with data? If so, we might need an if statement to make sure that `getMultisigIcpDetails` is finished

Reply:
this won't apply if https://github.com/cardano-foundation/cf-identity-wallet/pull/550#discussion_r1668740562 changes

Reply:
misread this, resolving